pkgs/by-name/ra/raylib/package.nix

```
@@ -1,85 +1,80 @@
-{ stdenv
-, lib
-, fetchFromGitHub
-, cmake
-, fetchpatch
-, mesa
-, libGLU
-, glfw
-, libX11
-, libXi
-, libXcursor
-, libXrandr
-, libXinerama
-, alsaSupport ? stdenv.hostPlatform.isLinux
-, alsa-lib
-, pulseSupport ? stdenv.hostPlatform.isLinux
-, libpulseaudio
-, sharedLib ? true
-, includeEverything ? true
-, raylib-games
-, darwin
-, autoPatchelfHook
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  cmake,
+  autoPatchelfHook,
+  glfw,
+  SDL2,
+  alsa-lib,
+  libpulseaudio,
+  raylib-games,
+  platform ? "Desktop", # Note that "Web", "Android" and "Raspberry Pi" do not currently work
+  pulseSupport ? stdenv.hostPlatform.isLinux,
+  alsaSupport ? false,
+  sharedLib ? true,
+  includeEverything ? true,
 }:
 let
-  inherit (darwin.apple_sdk.frameworks) Carbon Cocoa OpenGL;
-in
-stdenv.mkDerivation (finalAttrs: {
-  pname = "raylib";
-  version = "5.0";
+  inherit (lib) optional;
 
-  src = fetchFromGitHub {
-    owner = "raysan5";
-    repo = "raylib";
-    rev = finalAttrs.version;
-    hash = "sha256-gEstNs3huQ1uikVXOW4uoYnIDr5l8O9jgZRTX1mkRww=";
-  };
+  pname = "raylib";
+in
 
-  nativeBuildInputs = [
-    cmake
-  ] ++ lib.optional stdenv.hostPlatform.isLinux autoPatchelfHook;
+lib.checkListOfEnum "${pname}: platform"
+  [
+    "Desktop"
+    "Web"
+    "Android"
+    "Raspberry Pi"
+    "SDL"
+  ]
+  [ platform ]
+  (
+    stdenv.mkDerivation (finalAttrs: {
+      __structuredAttrs = true;
 
-  buildInputs = [ glfw ]
-    ++ lib.optionals stdenv.hostPlatform.isLinux [ mesa libXi libXcursor libXrandr libXinerama ]
-    ++ lib.optionals stdenv.hostPlatform.isDarwin [ Carbon Cocoa ]
-    ++ lib.optional alsaSupport alsa-lib
-    ++ lib.optional pulseSupport libpulseaudio;
+      inherit pname;
+      version = "5.5";
 
-  propagatedBuildInputs = lib.optionals stdenv.hostPlatform.isLinux [ libGLU libX11 ]
-    ++ lib.optionals stdenv.hostPlatform.isDarwin [ OpenGL ];
+      src = fetchFromGitHub {
+        owner = "raysan5";
+        repo = pname;
```

- https://github.com/NixOS/nixpkgs/issues/277994
```suggestion
        repo = "raylib";
```

---

Good to know, thanks